app/controllers/application_controller/buttons.rb

```diff
@@ -100,6 +100,36 @@ def group_update
   # we need a "Provider" so adding an "ExtManagementSystem" to the list.
   MODEL_WITH_OPEN_URL = %w[ExtManagementSystem Service User MiqGroup Tenant CloudTenant GenericObject Vm].freeze
 
+  # Types of visibility for the custom buttons
+  VISIBILITY_TYPES = {'role' => 'role', 'all' => 'all'}.freeze
+
+  def button_visibility_box_edit
+    typ_changed = params[:visibility_typ].present?
+    @edit[:new][:visibility_typ] = VISIBILITY_TYPES[params[:visibility_typ]] if typ_changed
+
+    visibility_typ = @edit[:new][:visibility_typ]
+    if %w[role].include?(visibility_typ)
```

Comment:
it's a single element array, perhaps we can simplify to:

```ruby
if visibility_typ == 'role'
```

Reply:
if `@edit[:new][:visibility_typ]` could be nil/empty, you could do:

```ruby
if visibility_typ.to_s == 'role'
```

Reply:
actually, the next line down assume `visibility_typ` is non-nil as it calls pluralize on it.  Can `@edit[:new][:visibility_typ]` be nil?

It looks like it can be nil if `VISIBILITY_TYPES[params[:visibility_typ]]` is nil?  Do we need to verify `params[:visibility_typ]` is valid and in the `VISIBILITY_TYPES` list?

Also, `@edit[:new][:visibility_typ]` could be unset if typ_changed is false. Are we guaranteed that it's set?